### ISSUE TITLE
RHEL 7.7 with selinux enabled using RHEL docker on k8s 1.16 cluster fails to deploy 

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#23662]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**
On a `rancher:2.3.2-rc2` - deploy an RHEL 7.7 custom cluster - (3 etcd, 1 control plane, 3 worker nodes)

Docker - Native docker 1.13
Selinux - Enabled

**Expected Result:**
The cluster should be deployed successfully

**Actual Result:**
The cluster fails to deploy with error: `Failed to get job complete status for job rke-network-plugin-deploy-job in namespace kube-system`

Error in logs:
```
2019/10/24 19:16:59 [INFO] Removing container [rke-log-cleaner] on host [xxxx], try #1


















2019/10/24 19:16:59 [INFO] cluster [c-xcdfw] provisioning: [cleanup] Successfully started [rke-log-cleaner] container on host [xxxx]
2019/10/24 19:16:59 [INFO] cluster [c-xcdfw] provisioning: [cleanup] Successfully started [rke-log-cleaner] container on host [xxxx]
2019/10/24 19:16:59 [INFO] Removing container [rke-log-cleaner] on host [xxxx], try rancher/rancher#1
2019/10/24 19:16:59 [INFO] cluster [c-xcdfw] provisioning: [remove/rke-log-cleaner] Successfully removed container on host [xxxxx]
2019/10/24 19:16:59 [INFO] cluster [c-xcdfw] provisioning: [remove/rke-log-cleaner] Successfully removed container on host [xxxx]
2019/10/24 19:16:59 [INFO] cluster [c-xcdfw] provisioning: [remove/rke-log-cleaner] Successfully removed container on host [xxxx]
2019/10/24 19:16:59 [INFO] cluster [c-xcdfw] provisioning: [remove/rke-log-cleaner] Successfully removed container on host [xxxx]
2019/10/24 19:16:59 [INFO] cluster [c-xcdfw] provisioning: [sync] Syncing nodes Labels and Taints
2019/10/24 19:16:59 [INFO] cluster [c-xcdfw] provisioning: [sync] Successfully synced nodes Labels and Taints
2019/10/24 19:16:59 [INFO] cluster [c-xcdfw] provisioning: [network] Setting up network plugin: canal
2019/10/24 19:17:00 [INFO] cluster [c-xcdfw] provisioning: [addons] Saving ConfigMap for addon rke-network-plugin to Kubernetes
2019/10/24 19:17:00 [INFO] cluster [c-xcdfw] provisioning: [addons] Successfully saved ConfigMap for addon rke-network-plugin to Kubernetes
2019/10/24 19:17:00 [INFO] cluster [c-xcdfw] provisioning: [addons] Executing deploy job rke-network-plugin
2019/10/24 19:17:35 [ERROR] cluster [c-xcdfw] provisioning: Failed to get job complete status for job rke-network-plugin-deploy-job in namespace kube-system
2019/10/24 19:17:35 [INFO] kontainerdriver rancherkubernetesengine stopped
2019/10/24 19:17:35 [ERROR] ClusterController c-xcdfw [cluster-provisioner-controller] failed with : Failed to get job complete status for job rke-network-plugin-deploy-job in namespace kube-system
```

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): 2.3.2-rc2
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): custom - rhel 7.7
- Kubernetes version (use `kubectl version`):

```
1.16.2-rancher1-1
```

- Docker version (use `docker version`):

```
native docker - docker 1.13 (native)
```

**Comment by kinarashah** [comment]
Error from the job, 
```
[ec2-user@ip-172-31-11-173 ~]$ kubectl --kubeconfig /tmp/kubeconfig_admin.yaml logs -f rke-network-plugin-deploy-job-xbchn -n kube-system

error: the path "/etc/config/rke-network-plugin.yaml" cannot be accessed: stat /etc/config/rke-network-plugin.yaml: permission denied
```

**Comment by sangeethah** [comment]
This issue is not seen when testing with K8s `1.15.5`

On a rancher:2.3.2-rc2 - deploy an RHEL 7.7 custom cluster - (3 etcd, 1 control plane, 3 worker nodes) using K8s  version `1.15.5` .
Docker - Native docker 1.13
Selinux - Enabled

**Comment by superseb** [comment]
Requested/provided more info on upstream issue: https://github.com/kubernetes/kubernetes/issues/83679

**Comment by sangeethah** [comment]
@sowmyav27 Could you test the case of not having rhel 7.7 nodes with Selinux not enabled and having  Native docker 1.13 for a k8s 1.16 and see if this issue is not seen ?

**Comment by sangeethah** [comment]
From @sowmyav27 testing , With RHEL 7.7 and selinux set to "permissive" , able to get a cluster is up and running usinf  k8s -1.16, native docker `1.13` using `v2.3.2-rc2`

**Comment by permanz** [comment]
testing with Centos7.7 with selinux disable and docker version 18.06.3-ce, system image: k8s -1.16.2， can not set up a cluster, the same issue "Failed to get job complete status for job rke-network-plugin-deploy-job in namespace kube-system"

**Comment by superseb** [comment]
This is fixed upstream and seems to be going to be backported.

**Comment by deniseschannon** [comment]
@superseb do they indicate when this would be backported and available?

**Comment by superseb** [comment]
No specific version, but based on https://github.com/kubernetes/kubernetes/pull/86036 probably v1.16.5

**Comment by superseb** [comment]
Backport PRs for 1.16 (https://github.com/kubernetes/kubernetes/pull/86036) and 1.17 (https://github.com/kubernetes/kubernetes/pull/86035) have been merged.

**Comment by fmunteanu** [comment]
@superseb is this issue still present? I'm going to test 2.3.4 with Selinux enabled and report back.

**Comment by fmunteanu** [comment]
@deniseschannon are these versions already part of RKE? Do I have to do any extra steps, while performing a [HA install](https://rancher.com/docs/rancher/v2.x/en/installation/ha/)?

**Comment by sangeethah** [comment]
@fmunteanu K8s version 16.6 should be already available to you on a fresh install of `v2.3.4`.
If you are using an already deployed rancher-server with `v2.3.4` you would have to refresh kubernetes metadata (Tools-> Refresh Kubernetes Metadata) if you dont see these new versions already.

**Comment by sowmyav27** [comment]
verified on 2.3.5-rc2,
On k8s version 1.16.6 and 1.17.2, cluster does not come up. On K8s version 1.15.9 --> Works fine

cluster spec:
```
deploy an RHEL 7.7 custom cluster - (3 etcd, 1 control plane, 3 worker nodes) 
Docker - Native docker 1.13
Selinux - Enabled
```
Error:
```2020/01/23 18:38:23 [ERROR] cluster [c-8klsx] provisioning: Failed to get job complete status for job rke-network-plugin-deploy-job in namespace kube-system
2020/01/23 18:38:23 [INFO] kontainerdriver rancherkubernetesengine stopped
2020/01/23 18:38:23 [ERROR] ClusterController c-8klsx [cluster-provisioner-controller] failed with : Failed to get job complete status for job rke-network-plugin-deploy-job in namespace kube-system
```

**Comment by fmunteanu** [comment]
Thanks for the tests @sowmyav27, it saved me time.

**Comment by superseb** [comment]
Upstream was not properly build, see https://github.com/kubernetes/kubernetes/issues/83679#issuecomment-578526051

**Comment by deniseschannon** [comment]
Based on this [comment](https://github.com/kubernetes/kubernetes/issues/83679#issuecomment-585197841) in k8s, k8s 1.16.8 and 1.17.4 should have this packaged correctly.

**Comment by fmunteanu** [comment]
Selinux is creating issues for Splunk logs also, into latest Rancher release, see https://github.com/rancher/rancher/issues/25182.

**Comment by deniseschannon** [comment]
Should be available with latest versions of k8s introduced: 1.17.4, 1.16.8

**Comment by sangeethah** [comment]
Tesed with latest build from master:

Able to provision a K8s `1.16.8` cluster on hosts with Rhel 7.7 and native docker 1.13 with selinux enabled with following configuration:
1 control
1 etcd
3 workers


From node Details page from Rancher UI
```
Docker Version: 1.13.1
Kubelet Version: v1.16.8
Kube Proxy Version: v1.16.8
OS: RHEL 7.7 3.10.0-1062.el7.x86_64
```

From host:
```
 #docker info | grep Registry
Registry: https://registry.access.redhat.com/v1/
 #docker info | grep "Server Version"
Server Version: 1.13.1
# hostnamectl | grep "Operating System"
  Operating System: Red Hat Enterprise Linux Server 7.7 (Maipo)
# getenforce
Enforcing
# sestatus | grep "SELinux status"
SELinux status:                 enabled
```

Able to provision a K8s `1.17.4` cluster on hosts with Rhel 7.7 and native docker 1.13 with selinux enabled with following configuration:
1 control
1 etcd
3 workers

From node Details page from Rancher UI
```
Docker Version: 1.13.1
Kubelet Version: v1.17.4
Kube Proxy Version: v1.17.4
OS: RHEL 7.7 3.10.0-1062.el7.x86_64
```
From host:
```
# docker info | grep Registry
Registry: https://registry.access.redhat.com/v1/
# docker info | grep "Server Version"
Server Version: 1.13.1
# hostnamectl | grep "Operating System"
  Operating System: Red Hat Enterprise Linux Server 7.7 (Maipo)
#  getenforce
Enforcing
# sestatus | grep "SELinux status"
SELinux status:                 enabled
```

**Comment by deniseschannon** [comment]
In `v2.3-head`, can we specifically validate this for v1.16.8-rancher1-1 and v1.17.4-rancher1-1. Since the templates between `master-head` and `v2.3-head` are different, we need to do a separate validation for this.

**Comment by sangeethah** [comment]
Tested with rancher server - `2.3.5` pointing to branch `dev-v2.3`.

Not ale to bring up a cluster successfully with K8s version `1.16.8`(1.16.8-rancher1) or `1.17.4` (v1.17.4-rancher1-1) using hosts with OS RHEL 7.7 , native docker 1.13 and SELINUX on.


Not able to bring up "canal" successfully.
`flexvol-driver` stuck in "Waiting" state with this error: 
`cp: can't create '/host/driver/.uds': Permission denied`

<img width="1294" alt="Screen Shot 2020-03-15 at 10 53 20 AM" src="https://user-images.githubusercontent.com/4266958/76707295-33a34400-66ab-11ea-8bcc-8890a4ad5408.png">

**Comment by superseb** [comment]
Available in `rancher/rancher:v2.3.6-rc6`

**Comment by sangeethah** [comment]
Tested with `rancher/rancher:v2.3.6-rc6`.

Able to bring up a cluster successfully with K8s version 1.16.8(1.16.8-rancher1) using hosts with OS RHEL 7.7 , native docker 1.13 and SELINUX on.
Automation runs succeed on this cluster. No new issues reported.

Able to bring up a cluster successfully with K8s version 1.17.4 (v1.17.4-rancher1-1)  using hosts with OS RHEL 7.7 , native docker 1.13 and SELINUX on.
Automation runs succeed on this cluster. No new issues reported.

Canal now uses `rancher/calico-node:v3.13.0`
<img width="1066" alt="Screen Shot 2020-03-16 at 4 17 10 PM" src="https://user-images.githubusercontent.com/4266958/76807464-adb5f480-67a1-11ea-911a-df0c83391d5b.png">
